kernel/typeops.ml

```diff
@@ -575,6 +575,10 @@ let type_of_global_in_context env r =
 
 let check_binder_annot s x =
   let r = x.binder_relevance in
+  let () = match r with
+  | Sorts.Relevant | Sorts.Irrelevant -> ()
+  | Sorts.RelevanceVar _ -> anomaly Pp.(str "the kernel does not support sort variables")
```

Comment:
Shouldn't there be a similar assert on case info's ci_relevance?

Reply:
It will break Typing because we use the same function there, and I believe this is currently dead code in the kernel because typing of the return sort should already prohibit variables to appear there.

Reply:
I suppose the worst case is that it gets handled by the relevance fixer

Reply:
I have added a dedicated check next to the only kernel call of check_case_info, just in case.